src/components/ImageView/index.native.js

```diff
@@ -56,6 +55,13 @@ class ImageView extends PureComponent {
         this.configureImageZoom = this.configureImageZoom.bind(this);
     }
 
+    componentDidUpdate(prevProps) {
+        if (this.props.url === prevProps.url || !this.interactionPromise) {
+            return;
+        }
+        this.state.interactionPromise.cancel();
+    }
+
```

Comment:
Explain this please

Reply:
if the url doesn't change from an update then the interactionPromise should continue unless there is no interactionPromise

Reply:
shouldn't the check should be `!this.state.interactionPromise` ?

If the `URL` changed then `interactionPromise` should happen for the new image right?

Reply:
yes if the url changes this cancel gets called, however if it's the same url the this interaction will be able to finish

Reply:
Why should this get cancelled?

Reply:
Because it’s loading the size of the previous image. I think you’re mixing up the logic a bit here if the previous url isn’t the same as the current url then it’s a completely different image and needs to load that